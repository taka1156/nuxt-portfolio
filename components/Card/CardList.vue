<template>
  <div class="CardLayout">
    <div class="cardlist">
      <div v-for="(cardInfo, index) in cards" :key="index" class="card">
        <card-layout :card-info="cardInfo" />
      </div>
    </div>
  </div>
</template>

<script>
import CardLayout from './CardLayout';

export default {
  name: 'CardLayout',
  components: {
    'card-layout': CardLayout
  },
  props: {
    cards: {
      type: Array,
      default: () => []
    }
  }
};
</script>

<style scoped>
.cardlist {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  flex-wrap: wrap;
  width: 90%;
  margin: 0 auto;
}

.card {
  width: 27rem;
  cursor: pointer;
  margin: 2em auto;
  border: solid 0.6px lightgray;
  border-radius: 1%;
  box-shadow: 5px 10px 20px rgba(0, 0, 0, 0.25);
}
</style>
