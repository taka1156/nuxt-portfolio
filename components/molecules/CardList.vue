<template>
  <div>
    <div v-if="cards.length !== 0" class="card-list">
      <card-item
        v-for="(cardInfo, index) in cards"
        :key="index"
        :card-info="cardInfo"
        class="card-list__card"
      ></card-item>
    </div>
  </div>
</template>

<script>
import CardListItem from '../molecules/CardListItem';

export default {
  name: 'CardList',
  components: {
    'card-item': CardListItem,
  },
  props: {
    cards: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
};
</script>

<style scoped>
.card-list {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  flex-wrap: wrap;
  width: 90%;
  margin: 0 auto;
}

.card-list__card {
  width: 27em;
  margin: 2.6em auto;
}
</style>
