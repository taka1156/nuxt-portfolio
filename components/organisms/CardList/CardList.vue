<template>
  <div>
    <div v-if="cards.length !== 0" class="card-list">
      <card-list-item
        v-for="card in cards"
        :key="card.name"
        :card="card"
        class="card-list__card"
      />
    </div>
  </div>
</template>

<script>
import CardListItem from '../../molecules/CardListItem/CardListItem';

export default {
  name: 'CardList',
  components: {
    CardListItem
  },
  props: {
    /**
     * カードリスト
     */
    cards: {
      type: Array,
      required: true
    }
  }
};
</script>

<style scoped>
.card-list {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 90%;
  margin: 0 auto;
}

.card-list__card {
  width: 25em;
  margin: 0 auto 2em;
}
</style>
