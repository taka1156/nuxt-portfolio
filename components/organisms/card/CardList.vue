<template>
  <div>
    <div v-if="cards.length !== 0" class="card-list">
      <card-list-item
        v-for="(card, index) in cards"
        :key="index"
        :card="card"
        class="card-list__card"
        @jump="jump"
      />
    </div>
  </div>
</template>

<script>
import CardListItem from '../../molecules/card/CardListItem';

export default {
  name: 'CardList',
  components: {
    'card-list-item': CardListItem,
  },
  props: {
    /**
     * カードリスト
     */
    cards: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
  methods: {
    /**
     * 制作物関連サイトに移動
     * (githubやwebサイトのリンク先)
     */
    jump(card = null) {
      if (card == null || card.link == null) {
        return;
      }
      location.assign(card.link);
    },
  },
};
</script>

<style scoped>
.card-list {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  flex-wrap: wrap;
  width: 90%;
  margin: 0 auto;
}

.card-list__card {
  width: 28em;
  margin: 0 auto 2em;
}
</style>
