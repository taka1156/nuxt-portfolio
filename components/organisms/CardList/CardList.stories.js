import CardList from './CardList.vue';
import { dummyCards, dummyCardsLink } from '@/@/__testdata__/testdata.js';

export default {
  title: 'Organisms/CardList',
  component: CardList
};

const Template1 = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { CardList },
  template: '<card-list v-bind="$props" />'
});

const Template2 = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { CardList },
  template: '<card-list v-bind="$props" />'
});

export const Default = Template1.bind({});
Default.args = { cards: dummyCards };

export const Link = Template2.bind({});
Link.args = { cards: dummyCardsLink };
