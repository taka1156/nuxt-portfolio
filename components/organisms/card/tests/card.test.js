import { mount } from '@vue/test-utils';
import CardList from '../CardList.vue';

// CardList
describe('CardList', () => {
  const cardList = propsData => {
    return mount(CardList, {
      propsData: {
        ...propsData,
      },
    });
  };

  const dummyCards = [
    {
      title: 'ダミータイトル',
      img: { url: 'http://placehold.jp/150x150.png' },
      content2: '詳細',
    },
    {
      title: 'ダミータイトル',
      img: { url: 'http://placehold.jp/150x150.png' },
      content2: '詳細',
    },
  ];

  const dummyCardsLink = [
    {
      title: 'ダミータイトル',
      img: { url: 'http://placehold.jp/150x150.png' },
      content2: '詳細',
      link: 'https://taka1156.site',
    },
    {
      title: 'ダミータイトル',
      img: { url: 'http://placehold.jp/150x150.png' },
      content2: '詳細',
      link: 'https://github.com/taka1156/nuxt-portoflio',
    },
  ];

  it('jumpメソッド', () => {
    // https://remarkablemark.org/blog/2018/11/17/mock-window-location/#update-for-jsdom-14
    delete window.location;
    window.location = { assign: jest.fn() };
    const wrapper = cardList({ cards: dummyCardsLink });
    wrapper.vm.jump(dummyCardsLink[0]);
    expect(window.location.assign).toBeCalledWith(dummyCardsLink[0].link);
  });

  it('CardListの初期値(linkなし): dummyCards', () => {
    const wrapper = cardList({ cards: dummyCards });
    console.log(wrapper.vm.$options.props.cards.default());
    expect(wrapper.vm.$options.props.cards.required).toBe(true);
    expect(wrapper.vm.cards).toBe(dummyCards);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('CardListの初期値(linkあり): dummyCardsLink', () => {
    const wrapper = cardList({ cards: dummyCardsLink });
    const btnTag = wrapper.findAll('button');
    expect(wrapper.vm.$options.props.cards.required).toBe(true);
    expect(btnTag.at(0).text()).toBe('WebSite');
    expect(btnTag.at(1).text()).toBe('GitHub');
    expect(wrapper.vm.cards).toBe(dummyCardsLink);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });
});
