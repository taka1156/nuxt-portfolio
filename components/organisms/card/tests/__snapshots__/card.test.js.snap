// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList CardListの初期値(linkあり): dummyCardsLink 1`] = `
"<div>
  <div class=\\"card-list\\">
    <div class=\\"card-list__card\\">
      <article class=\\"card-list-item\\">
        <figure><img alt=\\"ダミータイトルの画像\\" class=\\"card-img card-img--lg\\">
          <div class=\\"card-list-item__border\\"></div>
          <figcaption>
            <h2 class=\\"card-title\\">ダミータイトル</h2>
            <p class=\\"base-text base-text--card\\">詳細</p>
          </figcaption>
        </figure> <button class=\\"btn\\">
          WebSite
        </button>
      </article>
    </div>
    <div class=\\"card-list__card\\">
      <article class=\\"card-list-item\\">
        <figure><img alt=\\"ダミータイトルの画像\\" class=\\"card-img card-img--lg\\">
          <div class=\\"card-list-item__border\\"></div>
          <figcaption>
            <h2 class=\\"card-title\\">ダミータイトル</h2>
            <p class=\\"base-text base-text--card\\">詳細</p>
          </figcaption>
        </figure> <button class=\\"btn\\">
          GitHub
        </button>
      </article>
    </div>
  </div>
</div>"
`;

exports[`CardList CardListの初期値(linkなし): dummyCards 1`] = `
"<div>
  <div class=\\"card-list\\">
    <div class=\\"card-list__card\\">
      <article class=\\"card-list-item\\">
        <figure><img alt=\\"ダミータイトルの画像\\" class=\\"card-img card-img--sm\\">
          <div class=\\"card-list-item__border\\"></div>
          <figcaption>
            <h2 class=\\"card-title\\">ダミータイトル</h2>
            <p class=\\"base-text base-text--card\\">詳細</p>
          </figcaption>
        </figure>
        <!---->
      </article>
    </div>
    <div class=\\"card-list__card\\">
      <article class=\\"card-list-item\\">
        <figure><img alt=\\"ダミータイトルの画像\\" class=\\"card-img card-img--sm\\">
          <div class=\\"card-list-item__border\\"></div>
          <figcaption>
            <h2 class=\\"card-title\\">ダミータイトル</h2>
            <p class=\\"base-text base-text--card\\">詳細</p>
          </figcaption>
        </figure>
        <!---->
      </article>
    </div>
  </div>
</div>"
`;
